components/atom/validationCode/src/index.js

```diff
@@ -0,0 +1,67 @@
+import PropTypes from 'prop-types'
+import {useState, useRef, createRef} from 'react'
+
+const DEFAULT_LENGTH = 6
+const BASE_CLASS = 'react-AtomValidationCode'
+const noop = () => {}
+
+export default function AtomValidationCode({
+  onChange = noop,
+  label,
+  length = DEFAULT_LENGTH
+}) {
+  const [numbers, setNumbers] = useState(new Array(length).fill(null))
+
+  const numberRefs = useRef([])
+  numberRefs.current = numbers.map(
+    (_, i) => numberRefs.current[i] ?? createRef()
+  )
+
+  const handleChange = ({i, value}) => {
+    if (value < 10) {
+      const numberArray = [...numbers]
+      numberArray[i] = value
+      setNumbers(numberArray)
+      onChange(numberArray.join(''))
+    }
+
+    if (value !== '') {
+      const nextInput = numberRefs.current[i + 1]
+
+      nextInput
+        ? nextInput.current.focus()
+        : numberRefs.current[i].current.blur()
+    }
+  }
+
+  return (
+    <div className={BASE_CLASS}>
+      {label && <h3 className={`${BASE_CLASS}-label`}>{label}</h3>}
```

Comment:
IMHO this should not be inside the component. There is a special component `atom/label` that does this job.. Maybe we can keep it separated and let the developers compose it.